libs/providers/vendor-migration/src/lib/vendor-migration-provider.ts

```diff
@@ -0,0 +1,98 @@
+import { EvaluationContext, Provider, JsonValue, ResolutionDetails, DefaultLogger, Logger, OpenFeatureError } from '@openfeature/server-sdk';
+
+export class VendorMigrationProvider implements Provider {
+  metadata = {
+    name: VendorMigrationProvider.name,
+  };
+
+  readonly runsOn = 'server';
+
+  private readonly logger: Logger;
+
+  hooks = [];
+
+  constructor(private readonly providers: Provider[], private readonly flagEvaluationMode = "strict", logger?: Logger) {
```

Comment:
Ya I think we'll have to document this carefully as well. @luizgribeiro see [here](https://openfeature.dev/specification/sections/providers/#23-provider-hooks) about provider hooks.

I think you've have to add the hooks in the same order as the providers, and then expose that aggregated array.

There could be problems here though. Provider hooks are generally assumed to run as close to the evaluation as possible, and they might be making some changes to the context that cause issues for other providers. I don't think this would frequently be the case, but I could result in some troublesome behavior.

Reply:
If hooks become part of the evaluation process it'll probably be a good alternative to prevent context modifications leaking between providers.